app/controllers/payments_controller.rb

```diff
@@ -85,9 +85,10 @@ def webhook
     payment = Payment.find(payment_id)
     payment.update(status: Payment.order_to_payment_status(resource['status']))
 
-    # When fee is paid, user is granted membership
-    user = payment.user
-    user.grant_membership
+    # When member fee is paid, user is granted membership
```

Comment:
This means that the Controller has the responsibility for determining when membership should be granted membership (or not).  yuck.
This business logic belongs in a model.  Right now, we really don't have a model that takes care of this responsibility, so I'd vote for putting it into the User model for now.

This is another example of why we need to review our models and do some refactoring.  IMNSO, we definitely need a Service-type model (or maybe an ActiveModel) that has the responsibility of _setting the membership status for a person._

Reply:
OK, here's another approach, using the idea of "events" as discussed before (https://www.pivotaltracker.com/story/show/150343656).

The payment controller handles payments, and has no business deciding membership status.  This is an action that should occur post-payment processing (PP).

One way to do this is to create a PP event (in this case for a member payment) and take appropriate actions there.  This includes asking the User model to grant membership if appropriate (the basic logic I put in there reflects that we've _already_ decided to grant membership unless the user already has that .... probably need to make the decision in the user method itself).

So, events might suffice. A service class as you suggest would work also.  However, the Event class is kind of a generic service class that takes custom actions based on the specific event type.  I think we need an Event class is any case, for a number of reasons.

Lets discuss at our meeting today.